pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -884,6 +955,94 @@ func (r *Reconciler) updateHelmChartRefForGardenlets(ctx context.Context, log lo
 	return nil
 }
 
+func (r *Reconciler) reconcileDNSRecords(ctx context.Context, log logr.Logger, garden *operatorv1alpha1.Garden) error {
+	if garden.Spec.DNS == nil {
+		return fmt.Errorf("no DNS management configuration '.spec.dns' found in Garden resource")
+	}
+	if len(garden.Spec.DNS.Providers) == 0 {
+		return fmt.Errorf("no DNS providers specified at '.spec.dns.providers' in Garden resource")
+	}
```

Comment:
I think we can safely drop these checks because they are considered in the flow and validation.